### PR TITLE
[SEDONA-314] Support optimized join on ST_HausdorffDistance

### PR DESCRIPTION
**Pull request by iGN5117**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-314. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?
- Support optimized join on ST_Hausdorff by adding case matching on all overloaded hausdorff distances

## How was this patch tested?
- Added new unit tests

## Did this PR include necessary documentation updates?
- Yes, I have updated the documentation update.

### CODE REVIEW

File: sql/common/src/test/scala/org/apache/sedona/sql/predicateJoinTestScala.scala [hunk 1]
```diff
@@ -403,7 +403,41 @@ class predicateJoinTestScala extends TestBaseScala {
...
+      val inputPoint = buildPointDf.limit(sampleCount).repartition(5)
+      val inputPolygon = buildPolygonDf.limit(sampleCount).repartition(3)
 
+      distanceCandidates.foreach(distance => {
+
+        //DensityFrac specified, <= distance
+        val expectedDensityIntersects = bruteForceDistanceJoinHausdorff(sampleCount, distance, densityFrac, true)
+        val distanceDensityIntersectsDF = inputPoint.alias("pointDF").join(inputPolygon.alias("polygonDF"), expr(s"ST_HausdorffDistance(pointDF.pointshape, polygonDF.polygonshape, $densityFrac) <= $distance"))
+        assert(distanceDensityIntersectsDF.queryExecution.sparkPlan.collect { case p: DistanceJoinExec => p }.size === 1)
+        assert(distanceDensityIntersectsDF.count() == expectedDensityIntersects)
```
Comment (jiayuasu) [inline sql/common/src/test/scala/org/apache/sedona/sql/predicateJoinTestScala.scala:421]: I wonder what the number of expected intersects is in these test cases? Can you report the numbers here?

Comment (jiayuasu) [reply sql/common/src/test/scala/org/apache/sedona/sql/predicateJoinTestScala.scala:421]: @iGN5117

Comment (jiayuasu) [reply sql/common/src/test/scala/org/apache/sedona/sql/predicateJoinTestScala.scala:421]: Can you add more distanceCandidates? Most importantly, add some candidates that lead to results > 100.

Comment (iGN5117) [reply sql/common/src/test/scala/org/apache/sedona/sql/predicateJoinTestScala.scala:421]: Sure

Comment (iGN5117) [reply sql/common/src/test/scala/org/apache/sedona/sql/predicateJoinTestScala.scala:421]: Have added candidates 1, 2, 5, 10. These yield result 100, 298, 688, 1258 respectively for a sample size of 100

**Pull request merged**